### ISSUE TITLE
Comptime access violation crash

### ISSUE THREAD

**Issue by disarray2077** [beefytech/Beef#1948]
The comptime method below causes an access violation to the compiler when used: 

```bf
[Comptime]
static Type GetFuncPtrType<Class, MethodName>()
	where Class : class
	where MethodName : const String
{
	if (typeof(Class).IsGenericParam)
		return null;

	[Comptime]
	static String GetReturnCode()
	{
		if (typeof(Class).IsGenericParam)
			return null;

		let method = typeof(Class).GetMethod(MethodName).Get();

		String retType = method.ReturnType.GetFullName(.. scope .());

		String parameters = scope .();
		for (int i < method.ParamCount)
		{
			let paramType = method.GetParamType(i);
			parameters.Append(paramType.GetFullName(.. scope .()));

			if (i != method.ParamCount - 1)
				parameters.Append(", ");
		}

		return scope $"return typeof(function {retType}({parameters}));";
	}

	Compiler.Mixin(GetReturnCode());
	return null;
}
```

Upon inspection, it looks like this problem is due to `String.IsNullOrEmpty` not being generated for some reason:

![image](https://github.com/beefytech/Beef/assets/86157825/30bf65f6-5003-4eb2-99ef-47988c74b222)

and this only happens if you have a `String` extension like below:

```bf
namespace System
{
	extension String
	{
		public void Invert()
		{
			int i = 0;
			int j = mLength - 1;
			while (i < j)
			{
				let temp = Ptr[i];
				Ptr[i] = Ptr[j];
				Ptr[j] = temp;
				i++;
				j--;
			}
		}
	}
}
```

Tested with: f1a275d6efabfb7b2e2004295bfba3c030d98cf5

**Comment by disarray2077** [comment]
Looks like this works in another project 🤔, I will try to investigate more.

**Comment by bfiete** [comment]
Fixed at ab31e5bc89a0c4304d9716f9cb09614583ab5c3e
